values.go

```diff
@@ -327,6 +329,61 @@ func (n NullInt32) Encode(w *WriteBuf, oid Oid) error {
 	return encodeInt32(w, oid, n.Int32)
 }
 
+// Xid is PostgreSQL's Transaction ID type.
+//
+// In later versions of PostgreSQL, it is the type used for the backend_xid
+// and backend_xmin columns of the pg_stat_activity system view.
+//
+// Also, when one does
+//
+// 	select xmin, xmax, * from some_table;
+//
+// it is the data type of the xmin and xmax hidden system columns.
+//
+// It is currently implemented as an unsigned four byte integer.
+// Its definition can be found in src/include/postgres_ext.h as TransactionId
+// in the PostgreSQL sources.
+type Xid uint32
+
+// NullXid represents a Transaction ID (Xid) that may be null. NullXid implements the
+// Scanner and Encoder interfaces so it may be used both as an argument to
+// Query[Row] and a destination for Scan.
+//
+// If Valid is false then the value is NULL.
+type NullXid struct {
```

Comment:
But, yes, you inspired me to try to solve one of your root problems! :-)

On Sun, Sep 11, 2016 at 9:08 AM, Manni Wood <manni.lee.wood@gmail.com>
wrote:

> Another easy way it could happen in practice is if an xid-typed column is
> involved in a left join where no row can be joined to on the right-hand
> side of the join: that column will obviously satisfy the criteria: it will
> be type xid, and it will be null.
>
> On Sun, Sep 11, 2016 at 9:06 AM, James <notifications@github.com> wrote:
>
>> In values.go <https://github.com/jackc/pgx/pull/175#discussion_r78294810>
>> :
>>
>> > +//
>> > +// 	select xmin, xmax, * from some_table;
>> > +//
>> > +// it is the data type of the xmin and xmax hidden system columns.
>> > +//
>> > +// It is currently implemented as an unsigned four byte integer.
>> > +// Its definition can be found in src/include/postgres_ext.h as TransactionId
>> > +// in the PostgreSQL sources.
>> > +type Xid uint32
>> > +
>> > +// NullXid represents a Transaction ID (Xid) that may be null. NullXid implements the
>> > +// Scanner and Encoder interfaces so it may be used both as an argument to
>> > +// Query[Row] and a destination for Scan.
>> > +//
>> > +// If Valid is false then the value is NULL.
>> > +type NullXid struct {
>>
>> ha nice, yeah only places I could think of is where it is in a result but
>> uninitialized, and there it is pg_stat_activity. =)
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/jackc/pgx/pull/175/files/cba72d47c5f74ca539a66f2a37affb8878579d23#r78294810>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAFmKrcUUgDUSzihtO_iT5Ijm-RYOitAks5qo_ztgaJpZM4J1J84>
>> .
>>
>
>